### ISSUE TITLE
Windows test failure: org.eclipse.jetty.client.LivelockTest.testLivelock

### ISSUE THREAD

**Issue by joakime** [eclipse/jetty.project#2218]
```
java.lang.AssertionError
	at org.eclipse.jetty.client.LivelockTest.testLivelock(LivelockTest.java:131)

2018-02-22 00:12:44.553:INFO:oejs.Server:main: jetty-9.4.9-SNAPSHOT; built: 2018-02-22T00:00:24-07:00; git: e46459e8a89a7086b8578f83ab8e9b11e6306433; jvm 1.8.0_152-b16
2018-02-22 00:12:44.554:INFO:oejs.AbstractConnector:main: Started ServerConnector@a2ca681{HTTP/1.1,[http/1.1]}{0.0.0.0:60599}
2018-02-22 00:12:44.554:INFO:oejs.Server:main: Started @137123ms
2018-02-22 00:12:59.599:INFO:oejs.AbstractConnector:main: Stopped ServerConnector@a2ca681{HTTP/1.1,[http/1.1]}{0.0.0.0:0}
```

**Comment by joakime** [comment]
`jetty-9.4.x` branch

**Comment by joakime** [comment]
The individual clients do not connect and cannot countdown the latch.

```
result = {Result@1741} "Result[HttpRequest[GET /0 HTTP/1.1]@f053c2f > HttpResponse[null 0 null]@3cefa30d] java.net.SocketTimeoutException: Connect Timeout"
 request = {HttpRequest@1739} "HttpRequest[GET /0 HTTP/1.1]@f053c2f"
  headers = {HttpFields@1760} "Accept-Encoding: gzip\r\nUser-Agent: Jetty/9.4.z-SNAPSHOT\r\n\r\n"
  params = {Fields@1761} "{}"
  responseListeners = {ArrayList@1762}  size = 1
  aborted = {AtomicReference@1763} "java.net.SocketTimeoutException: Connect Timeout"
  client = {HttpClient@1764} 
  conversation = {HttpConversation@1738} "HttpConversation[64edc5aa]"
  host = "localhost"
  port = 62878
  uri = null
  scheme = "http"
  path = "/0"
  query = null
  method = "GET"
  version = {HttpVersion@1769} "HTTP/1.1"
  idleTimeout = 0
  timeout = 0
  timeoutAt = -1
  content = null
  followRedirects = true
  cookies = null
  attributes = null
  requestListeners = null
  pushListener = null
  trailers = null
 requestFailure = {SocketTimeoutException@1754} "java.net.SocketTimeoutException: Connect Timeout"
 response = {HttpResponse@1755} "HttpResponse[null 0 null]@3cefa30d"
  headers = {HttpFields@1776} "\r\n"
  request = {HttpRequest@1739} "HttpRequest[GET /0 HTTP/1.1]@f053c2f"
  listeners = {ArrayList@1762}  size = 1
  version = null
  status = 0
  reason = null
  trailers = null
 responseFailure = {SocketTimeoutException@1754} "java.net.SocketTimeoutException: Connect Timeout"
```

**Comment by joakime** [comment]
@sbordet any thoughts on this?

**Comment by sbordet** [comment]
@gregw after analysis with @joakime turns out that the test triggers a Windows JVM bug.

The test submits the self-submit action, then sends a request, which triggers a non-blocking connect that also submits the `Connect` action.
The `Connect` action registers for OP_CONNECT, but on Windows NIO never wakes up with `readyOps==OP_CONNECT` so the test fails for that reason with a connect timeout.

If the self-submit action is not submitted, the test passes on Windows because NIO does wake up with OP_CONNECT.

So it must be a combination of the continuous calls to `select()` caused by the self-submit action that triggers the fact that the OP_CONNECT event is not delivered by Windows NIO to Jetty.

Now, the problem is whether we may end up - in a real case, not a nasty test - in a situation that reproduces the behavior that triggers the JVM bug (e.g. see #2205).

We need to discuss what is the best solution to work around the Windows problem.

**Comment by joemokos** [comment]
@sbordet is there any update on this problem?  Our customer continues to experience 100% CPU.

**Comment by joakime** [comment]
@joemokos your issue and this issue (specific to jetty-client) are likely unrelated.

**Comment by joemokos** [comment]
@joakime Joakime, do you have a fix for this issue?  I would certainly like to give it a try since both seem related to Windows JVM NIO.  Also, in my case, the websocket connection is initiated by my application so it is using the jetty-client.

**Comment by joakime** [comment]
This is issue is about an intentional client livelock testcase.
And even that doesn't produce 100% CPU usage. the testcase just prevents OP_CONNECT from being detected in 1 out of 100,000 tests.
If you have an active connection, you made it past OP_CONNECT and are in a different scope of issue.

**Comment by joakime** [comment]
If the belief of @sbordet is true about a JVM bug, then there is little hope we can do work around it (as the bug would exist in the JVM's native selector code)

**Comment by joakime** [comment]
This has been fixed by commit 484280bac6956f9db0d98db60739f5368dec3691
